girder_worker/plugins/docker/executor.py

```diff
@@ -166,6 +167,27 @@ def _get_pre_args(task, uid, gid):
     return args
 
 
+def _create_named_output_pipes(task_outputs, outputs, tempdir):
+    """
+    Returns a map of output IDs to their named pipes, which are file descriptors
+    opened in read-only, non-blocking mode.
+    """
+    pipes = {}
+    for id, spec in task_outputs.iteritems():
+        if (spec.get('stream') and id in outputs and
+                spec.get('target') == 'filepath'):
+            path = spec.get('path', id)
+            if path.startswith('/'):
+                raise Exception('Streaming filepaths must be relative.')
+            path = os.path.join(tempdir, path)
+            os.mkfifo(path)
```

Comment:
I'd rather just put something into the module code in the docker plugin's `__init__.py` that raises an exception at startup time if the user isn't on Linux.